### PR TITLE
[lite-interpreter] speed_benchmark_torch support BUILD_LITE_INTERPRETER

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55402 [lite-interpreter] speed_benchmark_torch support BUILD_LITE_INTERPRETER**
```
Main run finished. Microseconds per iter: 40389.8. Iters per second: 24.7587
flame:/data/local/tmp $ ./speed_benchmark_torch --model=mobilenet2.ptl --input_type=float --input_dims=1,3,224,224 --warmup=5 --iter=20
Starting benchmark.
Running warmup runs.
Main runs.
Main run finished. Microseconds per iter: 41878.9. Iters per second: 23.8784
```
Differential Revision: [D27599824](https://our.internmc.facebook.com/intern/diff/D27599824)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -636,6 +636,10 @@ if(USE_VULKAN)
   endif()
 endif()
 
+if(BUILD_LITE_INTERPRETER)
+  string(APPEND CMAKE_CXX_FLAGS " -DBUILD_LITE_INTERPRETER")
```
Comment (cccclai) [inline CMakeLists.txt:640]: There exists an BUILD_LITE_INTERPRETER option in https://github.com/pytorch/pytorch/blob/8f78face63cb3a14d53325ec1ddcf5d5d110f796/CMakeLists.txt#L152

Comment (cccclai) [reply CMakeLists.txt:640]: Also I believe the macros is added here https://github.com/pytorch/pytorch/blob/2b31cddfe5c95e3523cf08c0b596a506c84642cd/caffe2/CMakeLists.txt#L1098

File: binaries/speed_benchmark_torch.cc [hunk 2]
```diff
@@ -21,6 +21,8 @@
 #include "caffe2/core/timer.h"
 #include "caffe2/utils/string_utils.h"
 #include <torch/csrc/autograd/grad_mode.h>
+#include <torch/csrc/jit/mobile/module.h>
```
Comment (cccclai) [inline binaries/speed_benchmark_torch.cc:24]: The include file is probably need be guarded by the macro `BUILD_LITE_INTERPRETER`

**Review by cccclai — COMMENTED** [review]
Thank you so much for the quick pr to enable speed_benchmark_torch with lite interpreter!

**Comment by iseeyuan** [comment]
Nice! In addition to @cccclai 's comments, I'm wondering if there's any CI covering this executable?

File: binaries/speed_benchmark_torch.cc (hunk 2 above)
Comment (IvanKobzarev) [reply binaries/speed_benchmark_torch.cc:24]: I checked 
`ANDROID_ABI=arm64-v8a ./scripts/build_android.sh -DBUILD_BINARY=ON` works ok with those includes

File: CMakeLists.txt (hunk 1 above)
Comment (IvanKobzarev) [reply CMakeLists.txt:640]: I checked commenting - it is not translated to binaries build

File: binaries/CMakeLists.txt [hunk 3]
```diff
@@ -4,7 +4,9 @@ if(INTERN_BUILD_MOBILE)
     caffe2_binary_target("speed_benchmark.cc")
   else()
     caffe2_binary_target("speed_benchmark_torch.cc")
-    caffe2_binary_target("compare_models_torch.cc")
+    if (NOT BUILD_LITE_INTERPRETER)
```
Comment (cccclai) [inline binaries/CMakeLists.txt:7]: if(NOT BUILD_LITE_INTERPRETER)

**Review by cccclai — APPROVED** [review]
Thanks for checking!  Fix the cmakelint and it's good to go.

**Pull request closed without merging**